utils/generate-rubocop-rules/src/main/java/org/sonarsource/slang/externalreport/rubocop/RuboCopRuleDefinition.java

```diff
@@ -0,0 +1,138 @@
+ */
+package org.sonarsource.slang.externalreport.rubocop;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.text.StringEscapeUtils;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class RuboCopRuleDefinition {
+
+  private static final Path RULES_FILE = Paths.get("sonar-ruby-plugin", "src", "main", "resources",
+    "org", "sonar", "l10n", "ruby", "rules", "rubocop", "rules.json");
+
+  private static final Path RUBOCOP_HELP = Paths.get("utils", "generate-rubocop-rules",
+    "src", "main", "resources", "rubocop.yml");
+
+  public static void main(String[] args) throws IOException {
+    String rules = RuboCopRuleDefinitionGenerator.generateRuleDefinitionJson(resolve(RUBOCOP_HELP));
+    Files.write(resolve(RULES_FILE), rules.getBytes(UTF_8));
+  }
+
+  private static Path resolve(Path path) throws IOException {
+    Path projectPath = Paths.get(".").toRealPath();
+    while (!projectPath.resolve(path).toFile().exists()) {
+      projectPath = projectPath.getParent();
+    }
+    return projectPath.resolve(path);
+  }
+
+  static final class RuboCopRuleDefinitionGenerator {
+
+    private RuboCopRuleDefinitionGenerator() {
+      // Utility class
+    }
+
+    static String generateRuleDefinitionJson(Path ruboCopHelpPath) throws IOException {
+      RuboCopHelp help = new RuboCopHelp(ruboCopHelpPath);
+      help.rules.sort(Comparator.comparing(a -> a.key));
+      Gson gson = new GsonBuilder().disableHtmlEscaping().setPrettyPrinting().create();
+      return gson.toJson(help.rules) + "\n";
+    }
+  }
+
+  private static class ExternalRule {
+    String key;
+    String name;
+    String type;
+    String severity;
+    String description;
+    String url;
+  }
+
+  private static class RuboCopHelp {
+
+    private static final String BASE_URL = "https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/";
+
+    private List<ExternalRule> rules;
+
+    private RuboCopHelp(Path ruboCopHelpPath) throws IOException {
+      this.rules = new ArrayList<>();
+      ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+      mapper.readTree(ruboCopHelpPath.toFile()).fields().forEachRemaining(this::addRule);
+    }
+
+    private void addRule(Map.Entry<String, JsonNode> entry) {
+      ExternalRule rule = new ExternalRule();
+      rule.key = entry.getKey();
+      String category = ruleCategory(rule.key);
+      String pascalCaseName = rule.key.substring(category.length() + 1).replaceAll("([^A-Z])([A-Z])", "$1 $2");
+      rule.name = pascalCaseName + " (" + category + ")";
```

Comment:
We could think of transforming this "category" to a tag. Just an idea.

Reply:
It could be an improvement ... once SQ platform support tags on external issue. Currently we can provide tags information on external issues, but we can not see them by tags in SQ UI.